File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2087.cpp
#include <iostream>
#include <algorithm>

using namespace std;

bool org(string a, string b){
    if(a.size() == b.size()){
        return a < b;
    }
    return a.size() < b.size();
}

int main(){
    int n;
    while(cin >> n){
        if(n == 0) break;
        
        string pal[100010];
        for(int i = 0; i < n; i++){
            cin >> pal[i];
        }
        
        sort(pal, pal+n, org);
        
        bool existe = true;
        for(int i = 0; i < n-1 && existe; i++){
            for(int j = i+1; j < n; j++){
                
                //cout << pal[i] << " " << pal[j] << endl;
                
                // percorre as palavras
                bool sub = true;
                for(int it = 0; it < pal[i].size(); it++){
                    
                    if(pal[i][it] != pal[j][it]){
                        //cout << pal[i][it] << " " << pal[j][it] << endl;
                        sub = false;
                        break;
                    }
                }
                
                if(sub){
                    existe = false;
                    break;
                }
                
            }
        }
    
        if(existe) cout << "Conjunto Bom" << endl;
        else cout << "Conjunto Ruim" << endl;
      
    }
    
    return 0;
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2087.py
while True:
    n = int(input())
    
    if n == 0: break
    
    pal = []
    
    for i in range(n):
        pal.append(input())
        
    pal = sorted(pal)
    
    existe = True
    for i in range(0, len(pal)-1):
        if not existe: break
        
        for j in range(i+1, len(pal)):
            
            #comparar palavras
            sub = True
            for it in range(0, min(len(pal[i]), len(pal[j]))):
                if(pal[i][it] != pal[j][it]):
                    sub = False
                    break
            
            if sub:
                existe = False
                break
    
    if existe: print("Conjunto Bom")
    else: print("Conjunto Ruim")